ddtrace/sampler.py

```diff
@@ -254,20 +254,21 @@ def __init__(
         if rate_limit is None:
             rate_limit = int(os.getenv("DD_TRACE_RATE_LIMIT", default=self.DEFAULT_RATE_LIMIT))
 
-        # Ensure rules is a list
-        self.rules = []  # type: List[SamplingRule]
         if rules is None:
             env_sampling_rules = os.getenv("DD_TRACE_SAMPLING_RULES")
             if env_sampling_rules:
                 rules = self._parse_rules_from_env_variable(env_sampling_rules)
             else:
                 rules = []
+            self.rules = rules
+        else:
+            self.rules = []
+            # Validate that rules is a list of SampleRules
+            for rule in rules:
+                if not isinstance(rule, SamplingRule):
+                    raise TypeError("Rule {!r} must be a sub-class of type ddtrace.sampler.SamplingRules".format(rule))
+                self.rules.append(rule)
 
-        # Validate that the rules is a list of SampleRules
-        for rule in rules:
```

Comment:
we were needlessly validating rules that we construct and know are SampleRules already